tests/Unit/Domain/Structure/Test_ElementId.cpp

```diff
@@ -165,6 +165,17 @@ void test_element_id() {
   // Test output operator:
   CHECK(get_output(block_2_3d) == "[B2,(L2I3,L1I0,L1I1)]");
   CHECK(get_output(element_six) == "[B4,(L0I0,L0I0,L0I0),G1]");
+  CHECK(ElementId<3>{"[B2,(L2I3,L1I0,L1I1)]"} == block_2_3d);
+  CHECK(ElementId<3>{"[B4,(L0I0,L0I0,L0I0),G1]"} == element_six);
+  CHECK(ElementId<1>{"[B1,(L2I3)]"} == ElementId<1>{1, {{{2, 3}}}});
+  CHECK(ElementId<1>{"[B1,(L2I3),G2]"} == ElementId<1>{1, {{{2, 3}}}, 2});
+  CHECK(ElementId<1>{"[B10,(L2I3)]"} == ElementId<1>{10, {{{2, 3}}}});
+  CHECK(ElementId<2>{"[B2,(L1I1,L2I0)]"} ==
+        ElementId<2>{2, {{{1, 1}, {2, 0}}}});
+  CHECK(ElementId<2>{"[B2,(L1I1,L2I0),G12]"} ==
+        ElementId<2>{2, {{{1, 1}, {2, 0}}}, 12});
+  CHECK(ElementId<2>{"[B52,(L12I133,L6I38)]"} ==
+        ElementId<2>{52, {{{12, 133}, {6, 38}}}});
```

Comment:
Optional: Add a couple of round-trip tests (`ElementId(get_output(id)) == id`) to make sure we don't change `operator<<` and forget to change the constructor to match.

Reply:
Good idea, done 👍